utils/oscap-ssh

```diff
@@ -80,61 +80,94 @@ function usage()
-    SSH_ADDITIONAL_ARGS="-t"
-    shift
-fi
-if [ $# -lt 2 ]; then
-    echo "Missing ssh host and ssh port."
-    usage
-    die
-fi
+# SSH_ADDITIONAL_OPTIONS may be defined in the calling shell
+SSH_TTY_ALLOCATION_OPTION=""
+
+# $1: The SSH command.
+# $2: More of additional options (optional, space-separated string)
+function ssh_execute_with_options {
+    ssh -o ControlPath="$MASTER_SOCKET" $SSH_ADDITIONAL_OPTIONS $2 -p "$SSH_PORT" "$SSH_HOST" "$1"
+}
+
+# $1: Local filename to copy
+# $2: Remote destination
+function scp_copy_to_temp_dir {
+    scp -o ControlPath="$MASTER_SOCKET" -P "$SSH_PORT" $SSH_ADDITIONAL_OPTIONS "$1" "$SSH_HOST:$REMOTE_TEMP_DIR/$2"
+}
+
+# $1: Remote filename to get
+# $2: Local destination
+function scp_retreive_from_temp_dir {
+    scp -o ControlPath="$MASTER_SOCKET" -P "$SSH_PORT" $SSH_ADDITIONAL_OPTIONS "$SSH_HOST:$REMOTE_TEMP_DIR/$1" "$2"
+}
+
+function sanity_check_arguments {
+    if [ $# -lt 1 ]; then
+        echo "No arguments provided."
+        usage
+        die
+    elif [ "$1" == "-h" ] || [ "$1" == "--help" ]; then
+        usage
+        die
+    elif [ "$1" == "sudo" ] || [ "$1" == "--sudo" ]; then
+        OSCAP_SUDO="sudo"
+        # force pseudo-tty allocation so that users can type their password if necessary
+        SSH_TTY_ALLOCATION_OPTION="-t"
+        shift
+    fi
+    if [ $# -lt 2 ]; then
+        echo "Missing ssh host and ssh port."
+        usage
+        die
+    fi
+}
+
+function check_oscap_arguments {
+    if [ "$1" == "--v" ] || [ "$1" == "--version" ]; then
+        true
+    elif [ "$1" == "-h" ] || [ "$1" == "--help" ]; then
+        true
+    elif [ "$1" == "info" ]; then
+        true
+    elif [ "$1 $2" == "xccdf eval" ]; then
+        true
+    elif [ "$1 $2" == "oval eval" ]; then
+        true
+    elif [ "$1 $2" == "oval collect" ]; then
+        true
+    else
+        die "This script only supports '-h', '--help', '--v', '--version', 'info', 'xccdf eval', 'oval eval' and 'oval collect'."
+    fi
+}
+
+sanity_check_arguments "$@"
 
 SSH_HOST="$1"
 SSH_PORT="$2"
 
-if [ "$3" == "--v" ] || [ "$3" == "--version" ]; then
-    true
-elif [ "$3" == "-h" ] || [ "$3" == "--help" ]; then
-    true
-elif [ "$3" == "info" ]; then
-    true
-elif [ "$3 $4" == "xccdf eval" ]; then
-    true
-elif [ "$3 $4" == "oval eval" ]; then
-    true
-elif [ "$3 $4" == "oval collect" ]; then
-    true
-else
-    die "This script only supports '-h', '--help', '--v', '--version', 'info', 'xccdf eval', 'oval eval' and 'oval collect'."
-fi
-
 shift 2
 
+check_oscap_arguments "$@"
+
 MASTER_SOCKET_DIR=$(mktemp -d)
 MASTER_SOCKET="$MASTER_SOCKET_DIR/ssh_socket"
 
 echo "Connecting to '$SSH_HOST' on port '$SSH_PORT'..."
-ssh -M -f -N -o ServerAliveInterval=60 -o ControlPath="$MASTER_SOCKET" -p "$SSH_PORT" "$SSH_HOST" || die "Failed to connect!"
+ssh -M -f -N -o ServerAliveInterval=60 -o ControlPath="$MASTER_SOCKET" -p "$SSH_PORT" $SSH_ADDITIONAL_OPTIONS "$SSH_HOST" || die "Failed to connect!"
```

Comment:
Could we perhaps add a wrapper for this command as well? I realize that it isn't reused and it adds more lines than necessary, but I think it'd be nice to have.